admin/plugins.php

```diff
@@ -0,0 +1,411 @@
+<?php
+/**
+ * Admin settings helper functions.
+ *
+ * @package performance-lab
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Gets plugin info from plugin slug.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $plugin_slug The string identifier for the plugin in questions slug.
+ *
+ * @return array Array of plugin data, or empty if none/error.
+ */
+function perflab_get_plugin_info( string $plugin_slug ) {
+	$plugin = plugins_api(
+		'plugin_information',
+		array(
+			'slug'   => $plugin_slug,
+			'fields' => array(
+				'short_description' => true,
+				'icons'             => true,
+			),
+		)
+	);
+
+	if ( is_wp_error( $plugin ) ) {
+		return array();
+	}
+
+	if ( is_object( $plugin ) ) {
+		$plugin = (array) $plugin;
+	}
+
+	return $plugin;
+}
+
+/**
+ * Gets info about all the standalone plugins and their status.
+ *
+ * @since n.e.x.t
+ *
+ * @return array of plugins info.
+ */
+function perflab_get_standalone_plugins() {
```

Comment:
Now that this function returns the data from WordPress.org as well, we should probably clarify that in the function name and docs too. For example we could rename it to `perflab_query_standalone_plugins_info()` or something like that?

Alternatively, we keep this function as you had it before to purely return the slugs, and we handle getting the WordPress.org API data directly within `perflab_render_plugins_ui()`. The latter would have the benefit of not introducing a function (yet) for this operation, so no public interface for something we don't know we'll need to have as a reusable piece of logic.

Reply:
@felixarntz I think decoupling it and moving it into the `perflab_render_plugins_ui()` will do for now.

Reply:
@felixarntz this has been addressed in https://github.com/WordPress/performance/pull/864/commits/f43d47ccdd5f820c918337d0213dcc853aacdceb